### PR TITLE
Bleve index directory preservation on upgrade

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation task: https://mattermost.atlassian.net/browse/MM-36754

Updated:
- Setup, Manage, Onboard, and Comply > Set Up Mattermost > Self-Managed Deployments > Configuration Settings > Bleve Settings (Experimental)
   - Added a Tip to Index Dir noting that the directory specified does not need to live within the ``mattermost``directory

### CODE REVIEW

File: source/administration/config-settings.rst [hunk 1]
```diff
@@ -5871,7 +5871,11 @@ Bleve Settings (Experimental)
 Index Dir
 ^^^^^^^^^^
 
-Directory path to use for storing bleve indexes.
+Directory path to use for storing bleve indexes. 
+
+.. tip::
+   
+   The Index directory path isn't required to exist within the ``mattermost`` directory. When specified outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade.
```
Comment (cwarnermm) [inline source/administration/config-settings.rst:5878]:
```suggestion
   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/administration/upgrade.html#upgrade-mattermost-server>`__ documentation for details.
```

File: source/administration/config-settings.rst [hunk 2]
```diff
@@ -5871,7 +5871,11 @@ Bleve Settings (Experimental)
 Index Dir
 ^^^^^^^^^^
 
-Directory path to use for storing bleve indexes.
+Directory path to use for storing bleve indexes. 
+
+.. tip::
+   
+   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/administration/upgrade.html#upgrade-mattermost-server>`__ documentation for details.
```
Comment (cwarnermm) [inline source/administration/config-settings.rst:5878]:
```suggestion
   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/administration/upgrade.html#upgrading-mattermost-server>`__ documentation for details.
```

File: source/administration/upgrade.rst [hunk 3]
```diff
@@ -89,14 +91,14 @@ Location of your local storage directory
 
 #. Remove all files *except data and custom directories* from within the current mattermost directory.
    
-    a. Run ``ls`` on your Mattermost install directory to identify what folders exist. If your folders match the structure below you can jump to step c.
+    a. Run ``ls`` on your Mattermost install directory to identify what default folders exist. If your folders match the structure specified in the note below, you can jump to step c.
 
-        - By default, your data directories will be preserved with the below commands. These are ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per *Before you begin*).
+        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<#before-you-begin>`_).
```
Comment (cwarnermm) [inline source/administration/upgrade.rst:96]:
```suggestion
        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/administration/upgrade.html#before-you-begin>`__).
```

File: source/administration/upgrade.rst [hunk 4]
```diff
@@ -89,14 +91,14 @@ Location of your local storage directory
 
 #. Remove all files *except data and custom directories* from within the current mattermost directory.
    
-    a. Run ``ls`` on your Mattermost install directory to identify what folders exist. If your folders match the structure below you can jump to step c.
+    a. Run ``ls`` on your Mattermost install directory to identify what default folders exist. If your folders match the structure specified in the note below, you can jump to step c.
 
-        - By default, your data directories will be preserved with the below commands. These are ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per *Before you begin*).
+        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/administration/upgrade.html#before-you-begin>`__).
```
Comment (cwarnermm) [inline source/administration/upgrade.rst:96]:
```suggestion
        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__).
```

File: source/deployment/bleve.rst [hunk 5]
```diff
@@ -14,15 +14,15 @@ Follow these steps to configure the Mattermost server to use Bleve and generate
...
 6. Set **Enable Bleve for search queries** to **true**.
 7. Set **Enable Bleve for autocomplete queries** to **true**.
 
 .. note::
 
-  Search results for files shared before upgrading to Mattermost Server 5.35 may be incomplete until an `extraction command <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-extract-documents-content>`__ is executed using the CLI. After running this command, the search index must be rebuilt. Go to **System Console > Experimental > Bleve > Bulk Indexing**, then select **Index Now** to rebuild the search index to include older file contents.
+  Search results for files shared before upgrading to Mattermost Server v5.35 may be incomplete until an `extraction command <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-extract-documents-content>`__ is executed using the CLI. After running this command, the search index must be rebuilt. Go to **System Console > Experimental > Bleve > Bulk Indexing**, then select **Index Now** to rebuild the search index to include older file contents.
```
Comment (cwarnermm) [inline source/deployment/bleve.rst:25]:
```suggestion
  Search results for files shared before upgrading to Mattermost Server v5.35 may be incomplete until an `extraction command <https://docs.mattermost.com/manage/command-line-tools.html#mattermost-extract-documents-content>`__ is executed using the CLI. After running this command, the search index must be rebuilt. Go to **System Console > Experimental > Bleve > Bulk Indexing**, then select **Index Now** to rebuild the search index to include older file contents.
```

File: source/administration/upgrade.rst [hunk 6]
```diff
@@ -89,14 +91,14 @@ Location of your local storage directory
...
+        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__).
         - Custom directories are directories that you've added to Mattermost and are not preserved by default. Generally, these are TLS keys or other custom information.
+        - If using `Bleve Search <https://docs.mattermost.com/deployment/bleve.html#bleve-search-experimental>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading, or following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory, then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/administration/config-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```
Comment (cwarnermm) [inline source/administration/upgrade.rst:98]:
```suggestion
        - If using `Bleve Search <https://docs.mattermost.com/deployment/bleve.html#bleve-search-experimental>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading, or following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory, then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```

File: source/administration/upgrade.rst [hunk 7]
```diff
@@ -89,14 +91,14 @@ Location of your local storage directory
...
+        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__).
         - Custom directories are directories that you've added to Mattermost and are not preserved by default. Generally, these are TLS keys or other custom information.
+        - If using `Bleve Search <https://docs.mattermost.com/deployment/bleve.html#bleve-search-experimental>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading, or following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory, then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```
Comment (cwarnermm) [inline source/administration/upgrade.rst:98]:
```suggestion
        - If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading, or following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory, then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```

File: source/deployment/bleve.rst [hunk 8]
```diff
@@ -14,15 +14,15 @@ Follow these steps to configure the Mattermost server to use Bleve and generate
 
 1. Open **System Console > Experimental > Bleve**.
 2. Set **Enable Bleve Indexing** to **true** to enable the other settings on the page.
-3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory.
+3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory. See our `Configuration Settings <https://docs.mattermost.com/administration/config-settings.html#bleve-settings-experimental>`__  documentation for details.
```
Comment (cwarnermm) [inline source/deployment/bleve.rst:17]:
```suggestion
3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__  documentation for details.
```

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @cwarnermm! This is going to be useful.

File: source/configure/configuration-settings.rst [hunk 9]
```diff
@@ -5885,7 +5885,11 @@ Bleve Settings (Experimental)
 Index Dir
 ^^^^^^^^^^
 
-Directory path to use for storing bleve indexes.
+Directory path to use for storing bleve indexes. 
+
+.. tip::
+   
+   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/administration/upgrade.html#upgrading-mattermost-server>`__ documentation for details.
```
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:5892]:
```suggestion
   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__ documentation for details.
```

File: source/configure/configuration-settings.rst [hunk 10]
```diff
@@ -5885,7 +5885,11 @@ Bleve Settings (Experimental)
 Index Dir
 ^^^^^^^^^^
 
-Directory path to use for storing bleve indexes.
+Directory path to use for storing bleve indexes. 
+
+.. tip::
+   
+   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, additional steps aren't needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__ documentation for details.
```
Comment (justinegeffen) [inline source/configure/configuration-settings.rst:5892]:
```suggestion
   The bleve index directory path isn't required to exist within the ``mattermost`` directory. When it exists outside of the ``mattermost`` directory, no  additional steps are needed to preserve or reindex these files as part of a Mattermost upgrade. See our `Upgrading Mattermost Server <https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__ documentation for details.
```

File: source/deploy/bleve-search.rst [hunk 11]
```diff
@@ -14,15 +14,15 @@ Follow these steps to configure the Mattermost server to use Bleve and generate
 
 1. Open **System Console > Experimental > Bleve**.
 2. Set **Enable Bleve Indexing** to **true** to enable the other settings on the page.
-3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory.
+3. Set the directory path to use for storing bleve indexes (e.g.: ``/var/opt/mattermost/bleveindexes``). The user running Mattermost should have permissions to access the directory. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__  documentation for details.
 4. Save the configuration.
-5. Click the **Index Now** button. All users, channels, and posts in the database will be indexed oldest to newest.
+5. Select the **Index Now** button. All users, channels, and posts in the database will be indexed oldest to newest.
```
Comment (justinegeffen) [inline source/deploy/bleve-search.rst:19]:
```suggestion
5. Select **Index Now**. All users, channels, and posts in the database will be indexed oldest to newest.
```

File: source/upgrade/upgrading-mattermost-server.rst [hunk 12]
```diff
@@ -89,14 +91,14 @@ Location of your local storage directory
...
+        - By default, your data directories will be preserved with the commands below, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage, as per `<https://docs.mattermost.com/upgrade/upgrading-mattermost-server.html>`__).
         - Custom directories are directories that you've added to Mattermost and are not preserved by default. Generally, these are TLS keys or other custom information.
+        - If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading, or following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory, then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```
Comment (justinegeffen) [inline source/upgrade/upgrading-mattermost-server.rst:98]:
```suggestion
        - If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, the index directory path won't be preserved with the commands below if the directory exists within the ``mattermost`` directory. You can either move the bleve index directory out from the ``mattermost`` directory before upgrading or, following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory. You can then store that directory or re-index. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```

**Review by justinegeffen — APPROVED** [review]
Some minor comments that aren't blocking but may enhance the overall content. Approving for expediency - LMK if you have any questions though!

**Pull request merged**